src/models/DataSource.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * LINKURIOUS CONFIDENTIAL
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-09-02.
+ */
+import { GenericObject, ICaptionsConfig } from './Configuration';
```

Comment:
Why using interfaces and not types ?

Reply:
What would be the benefit here?

Reply:
Just consistency and maintenability. Interfaces should be used for classes.

Reply:
when possible we use interfaces, because they can be used with implements and extends keywords and they work exactly as a `type` for the rest.

we define it a `type` only when needed for more complex types that couldn't be phrased as an interface and therefore cannot be used with the implements and extends keywords.

Reply:
@maxime-allex @Diachk open to discuss, but I don't see any gain in transforming an interface into a type.

Reply:
You can extends a type to by using `type Type = OtherType + {...}`.

For `implement`, I agree, but you use that keyword with a Class.

That's just a question of taste. I feel more confortable when usages are simple and clearly defined.

Anyway, that's your repo so do whatever you want 😛

Reply:
CaptionConfig is a class in Linkurious server that implements ICaptionConfig, so it makes sense for us to define it as an interface. Except maybe code style or personal preference, i don't see additional benefit as well

Reply:
> CaptionConfig is a class in Linkurious server that implements ICaptionConfig, so it makes sense for us to define it as an interface

Yes

> Except maybe code style or personal preference

Yes, that's what I said below.